src/clj_kondo/impl/analyzer.clj

```diff
@@ -934,6 +934,21 @@
 (defn- let? [x]
   (one-of x [[clojure.core let] [cljs.core let]]))
 
+(defn- invalid-fn-name? [?name-expr]
+  (let [valid? (some-fn utils/list-node? ; fn body (usually 1 of many)
+                        utils/vector-node? ; fn args
+                        utils/symbol-token?)] ; fn name
+    (not (valid? ?name-expr))))
+
+(defn- reg-invalid-fn-name! [ctx expr filename]
+  (findings/reg-finding!
+   ctx
+   (node->line
+    filename
+    expr
+    :invalid-fn-name
+    "First arg of fn should be a symbol, arg vector or body list")))
```

Comment:
"body list" isn't quite right. I'm not sure what we call the (multiple) parenthesized lists... arities?